lib/spack/spack/main.py

```diff
@@ -123,30 +122,23 @@ def add_all_commands(parser):
 def get_version():
     """Get a descriptive version of this instance of Spack.
 
-    If this is a git repository, and if it is not on a release tag,
-    return a string like:
-
-        release_version-commits_since_release-commit
-
-    If we *are* at a release tag, or if this is not a git repo, return
-    the real spack release number (e.g., 0.13.3).
+    Outputs '<PEP440 version> (<git commit sha>)'.
 
+    The commit sha is only added when available.
     """
+    version = spack.spack_version
     git_path = os.path.join(spack.paths.prefix, ".git")
     if os.path.exists(git_path):
         git = exe.which("git")
-        if git:
-            with fs.working_dir(spack.paths.prefix):
-                desc = git("describe", "--tags", "--match", "v*",
-                           output=str, error=os.devnull, fail_on_error=False)
-
-            if git.returncode == 0:
-                match = re.match(r"v([^-]+)-([^-]+)-g([a-f\d]+)", desc)
-                if match:
-                    v, n, commit = match.groups()
-                    return "%s-%s-%s" % (v, n, commit)
-
-    return spack.spack_version
+        if not git:
+            return version
+        rev = git('-C', spack.paths.prefix, 'rev-parse', 'HEAD',
+                  output=str, error=os.devnull, fail_on_error=False)
+        match = re.match(r"[a-f\d]{7,}$", rev)
+        if match:
+            version += " ({0})".format(match.group(0))
```

Comment:
This should still be guarded by whether git returned correctly:

```suggestion
        if git.returncode == 0:
            match = re.match(r"[a-f\d]{7,}$", rev)
            if match:
                version += " ({0})".format(match.group(0))
```

If something goes wrong with `rev-parse` or if git is just broken, I don't know what that output will look like and I don't think we should try to match it.  Just return the version as if it wasn't a git repo in that case.